### PR TITLE
Add realtime gcCycleOn fields to OMR_VM from OpenJ9

### PR DESCRIPTION
**Pull request by jason-hall**
Signed-off-by: Jason Hall <jasonhal@ca.ibm.com>

### CODE REVIEW

File: include_core/omr.h [hunk 1]
```diff
@@ -143,6 +143,10 @@ typedef struct OMR_VM {
 	struct OMRTraceEngine *_trcEngine;
 	void *_methodDictionary;
 #endif /* OMR_RAS_TDF_TRACE */
+#endif /* OMR_RAS_TDF_TRACE */
+	omrthread_monitor_t _gcCycleOnMonitor;
+	uintptr_t _gcCycleOn;
+#if defined(OMR_GC_REALTIME)
```
Comment (youngar) [inline include_core/omr.h]:
```suggestion
#endif /* defined(OMR_GC_REALTIME) */
```

File: gc/base/GCExtensionsBase.cpp [hunk 2]
```diff
@@ -180,6 +180,13 @@ MM_GCExtensionsBase::initialize(MM_EnvironmentBase* env)
 		goto failed;
 	}
 
+#if defined(OMRVM_GC_REALTIME)
```
Comment (youngar) [inline gc/base/GCExtensionsBase.cpp]: Another small thing I just noticed, this code is using two different flags `OMRVM_GC_REALTIME` and `OMR_GC_REALTIME`.  Do you think we could add one of these flags and enable them while this work is ongoing?

Comment (jason-hall) [reply gc/base/GCExtensionsBase.cpp]: Typo

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by charliegracie** [comment]
@youngar once you approve feel free to merge

**Pull request merged**